src/app/components/Account/index.jsx

```diff
@@ -0,0 +1,154 @@
+import React, { useState, useEffect } from "react";
+import { useLocation, useHistory } from "react-router-dom";
+
+import { Form, Row, Col, Menu, Dropdown, Button, message } from "antd";
+
+import accountSvc from "../../../common/services/account.svc";
+import connectors from "../../../extension/background-script/connectors";
+import LndForm from "../Lnd";
+import LndHubForm from "../LndHub";
+import LnBitsForm from "../LnBits";
+import NativeConnectionForm from "../NativeConnection";
+
+const CONNECTORS = {
+  lnd: "LND",
+  lndHub: "LndHub",
+  lnBits: "LnBits",
+  native: "Native",
+};
+
+const Account = () => {
+  const history = useHistory();
+  const location = useLocation();
+  const accountId = location.state && location.state.accountId;
+  const [account, setAccount] = useState({});
+  const [accountType, setAccountType] = useState("lnd");
+
+  useEffect(() => {
+    async function fetchAccount() {
+      const acc = (await accountSvc.getById(accountId)) || {};
+      setAccount(acc);
+      if (acc.type) {
+        setAccountType(acc.type);
+      }
+    }
+    fetchAccount();
+  }, [accountId]);
+
+  let connectorForm = null;
+  const submitHook = (innerForm) => {
+    connectorForm = innerForm;
+  };
+
+  const handleSubmit = async () => {
+    try {
+      const values = connectorForm && (await connectorForm.validateFields());
+      const newAccount = Object.assign({}, account || {}, values || {});
+      newAccount.type = accountType;
+      setAccount(newAccount);
+      if (newAccount.id) {
+        await accountSvc.update(newAccount);
+        message.success(`Account ${newAccount.name} updated!`);
+      } else {
+        await accountSvc.add(newAccount);
+        message.success(`Account ${newAccount.name} created!`);
+      }
+      history.goBack();
+    } catch (err) {
+      message.error("Please check field values!");
+    }
+  };
+
+  const handleConnectorTypeChange = (type) => {
+    setAccountType(type);
+  };
+
+  const handleTestAccount = async () => {
+    try {
+      const values = connectorForm && (await connectorForm.validateFields());
+      const lndConnector = new connectors.lnd({
```

Comment:
so far I did not expect the connectors to be called from the UI, but only from the background script. Not sure if that always works? 

I would see testing the connection as part of some validation. (then the user also does not need to extra "klick" test.
In Johns [UI suggestion](https://www.figma.com/file/xEPJFGFdc3HD0JS1ax2IpH/Lightning-Browser-Extension?node-id=1%3A22152) we have two steps: first the form, then the confirmation. 
what do you think about that?

Reply:
agree